hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -240,14 +241,23 @@ public static <T extends Throwable> T tryCreateExceptionWithMessageAndCause(Clas
     private static <T extends Throwable> T cloneException(Class<? extends Throwable> exceptionClass,
                                                           String message, @Nullable Throwable cause,
                                                           ConstructorMethod constructorMethod) {
+        Constructor<? extends Throwable> reflectedConstructor = null;
+        boolean isAccessible = false;
         try {
-            MethodHandle constructor = MethodHandles.publicLookup()
-                    .findConstructor(exceptionClass, constructorMethod.signature());
+            reflectedConstructor = exceptionClass.getDeclaredConstructor(constructorMethod.signature().parameterArray());
```

Comment:
i'm not sure whether it is good practice to create an object by using its non-public constructor, there can be a reason to restrict access to it.

Reply:
In general, I agree. But we here hack this nevertheless, by artifically manipulating the stacktrace.

Alternatively we can wrap should exception in some generic one, WDYT?

Reply:
yes, wrapping seems better to me.

Reply:
I went with just sending original exception; detecting non-public constructor and wrapping would only make code messier, IMHO not worth in the end